From: Jiri Slaby <jslaby@suse.cz>
Date: Mon, 24 Apr 2017 08:52:26 +0200
Subject: Revert "acpi, nfit, libnvdimm: fix interleave set cookie calculation
 (64-bit comparison)"
Patch-mainline: never, kabi
References: kabi

This reverts commit 074bcc1302fd4357fa30c167bb20f684998b025f, upstream
commit b03b99a329a14b7302f37c3ea6da3848db41c8c5. It fixes commit
86ef58a4e35e8fa66afb5898cf6dec6a3bb29f67, but we have reverted that one
for kABI reasons in:
patches.kabi/revert-nfit-libnvdimm-fix-interleave-set-cookie-calc.patch

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 drivers/acpi/nfit.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/drivers/acpi/nfit.c b/drivers/acpi/nfit.c
index 67d7489ced01..14c2a07c9f3f 100644
--- a/drivers/acpi/nfit.c
+++ b/drivers/acpi/nfit.c
@@ -979,11 +979,7 @@ static int cmp_map(const void *m0, const void *m1)
 	const struct nfit_set_info_map *map0 = m0;
 	const struct nfit_set_info_map *map1 = m1;
 
-	if (map0->region_offset < map1->region_offset)
-		return -1;
-	else if (map0->region_offset > map1->region_offset)
-		return 1;
-	return 0;
+	return map0->region_offset - map1->region_offset;
 }
 
 /* Retrieve the nth entry referencing this spa */
-- 
2.12.2

